kedro/config/common.py

```diff
@@ -147,11 +155,14 @@ def _load_config_file(config_file: Path, ac_template: bool = False) -> Dict[str,
 
     try:
         # Default to UTF-8, which is Python 3 default encoding, to decode the file
-        with open(config_file, encoding="utf8") as yml:
+        with open(config_file, encoding="utf8") as file:
             _config_logger.debug("Loading config file: '%s'", config_file)
             return {
                 k: v
-                for k, v in anyconfig.load(yml, ac_template=ac_template).items()
+                for k, v in anyconfig.load(
+                    file,
+                    **(anyconfig_args if anyconfig_args else DEFAULT_ANYCONFIG_ARGS),
```

Comment:
I don't think we need this logic in every function - just the top level one  that actually gets called.